### ISSUE TITLE
Explicit representation of volumes

### ISSUE THREAD

**Issue by mjpost** [acl-org/acl-anthology#2379]
Our journal-handling is hard-coded, that is, we have [a list of journals in data.py](https://github.com/acl-org/acl-anthology/blob/dae3430d53085bfea11491b005f4907375702dbe/bin/anthology/data.py#L72), and when we generate the bibtex, we [parse the booktitle](https://github.com/acl-org/acl-anthology/blob/dae3430d53085bfea11491b005f4907375702dbe/bin/anthology/volumes.py#L135) to figure out volume and issue numbers.

There are a number of issues with this; for one, we have other journals (e.g., [LILT](https://aclanthology.org/venues/lilt/)) that are not currently processed as such. Moreover, we should move this out of the code and into the XML, and adapt the code accordingly.

We should add explicit `<volume-number>` and `<issue>` tags (names might change) to the XML, moving it out of the code, for all journals.

_Originally posted by @mjpost in https://github.com/acl-org/acl-anthology/issues/2339#issuecomment-1422694257_

**Comment by mjpost** [comment]
CC @mbollmann @akoehn in case you have comments (or time/interest to help)

**Comment by mbollmann** [comment]
I wonder if we should call it something like `<bibvolume>` to better avoid confusion with our internal `<volume>`.

**Comment by mjpost** [comment]
How do you feel about `<volume-issue>` and `<volume-number>`?

**Comment by mbollmann** [comment]
Yeah, we're still overloading the term "volume" here, but this probably makes it clear enough. We could do `<bib-volume-{issue,number}>` but maybe that's just verbose for no good reason.

**Comment by akoehn** [comment]
I think those two are good, especially when their semantics is documented in our schema file (as a comment). Anyone wanting to understand the data structure in more detail should have a look at the schema anyway.

**Comment by mjpost** [comment]
Honestly I'm kind of tempted to suggest we rename the current "volume" to "proceedings" but I suspect it would take a lot of reworking.

**Comment by mbollmann** [comment]
> Honestly I'm kind of tempted to suggest we rename the current "volume" to "proceedings" but I suspect it would take a lot of reworking.

Search and replace might get us quite far here actually. ;) Although we publicize `/volumes/` URLs, so that's one part that probably shouldn't change.

**Comment by mbollmann** [comment]
@mjpost
> How do you feel about `<volume-issue>` and `<volume-number>`?

While trying to implement this, I found these names to be incredibly confusing. For example, BibTeX's `number` field corresponds to CSL's `issue` field, and in practice (see below for examples), different journals use either "number" or "issue" for exactly the same thing. So I feel strongly now that we shouldn't use both, and maybe accept the inconvenience of naming one of them `<bib-volume>` or `<journal-volume>`.

A naming convention like `<journal-volume>` and `<journal-issue>` would also indicate that these tags are only supposed to be used for journal publications.

However, I also stumbled over some other questions:

### 1. What should be in the `<booktitle>` field of a journal?

For [2022.cl-1](https://aclanthology.org/volumes/2022.cl-1/), the XML currently has:

```xml
<booktitle>Computational Linguistics, Volume 48, Issue 1 - March 2022</booktitle>
```

Currently, [`data.get_journal_info()`](https://github.com/acl-org/acl-anthology/blob/9714206771c01f9612d1db2024c5d559b752dfac/bin/anthology/data.py#L124) contains hard-coded if-statements that map this to the "pure" journal title, in this case `Computational Linguistics`. This is used e.g. to populate the `journal` field in the BibTeX entries for the individual articles. If we want to get rid of data mixed into the code, this journal title needs to be represented explicitly in the XML.

If we introduce explicit volume number and issue representations, everything else in that field is redundant (month/year already have their own tags), so we could do this:

```xml
<booktitle>Computational Linguistics</booktitle>
<journal-volume>48</journal-volume>
<journal-issue>1</journal-issue>
<month>March</month>
<year>2022</year>
```

However, that raises two issues:

#### a. Formatting of the full journal title will change.

Here are some journal `<booktitle>`s that we currently have:

- [Computational Linguistics, Volume 48, Issue 1 - March 2022](https://aclanthology.org/volumes/2022.cl-1/)
- [International Journal of Computational Linguistics & Chinese Language Processing, Volume 23, Number 1, June 2018](https://aclanthology.org/volumes/2018.ijclclp-1/)
- [Northern European Journal of Language Technology, Volume 8](https://aclanthology.org/volumes/2022.nejlt-1/)
- [American Journal of Computational Linguistics (February 1977)](https://aclanthology.org/volumes/J77-1/)

Notice how some say "Issue", others "Number"; different separators are used; some include the month/year of publication in their booktitle, others don't; and a few only have month/year, but no number at all. If we move to a representation like suggested above and generate these "full titles" automatically from the data, they won't look identical to how they look now.

#### b. Some journals have subtitles.

How should a journal issue like the following be represented?

- [Traitement Automatique des Langues, Volume 52, Numéro 2 : Vers la morphologie et au-delà [Toward Morphology and beyond]](https://aclanthology.org/volumes/2011.tal-2/)

Like I motivated above, we want the journal's name `Traitement Automatique des Langues` to be explicitly represented in the XML, but then where do we put the subtitle? One way might be to introduce a new XML tag for this:

```xml
<booktitle>Traitement Automatique des Langues</booktitle>
<booksubtitle>Vers la morphologie et au-delà [Toward Morphology and beyond]</booksubtitle>
<journal-volume>52</journal-volume>
<journal-issue>2</journal-issue>
```

I'm really wary of introducing yet another XML tag, but I would also like to avoid keeping the full, redundant `<booktitle>` field around that we have now.

### 2. How do we indicate that something _is_ a journal in the XML?

It's important to know that something is a journal vs. a conference proceedings e.g. for generating the BibTeX. Where do we put that in the XML?

My intuition is to use an _attribute_ since we'd have a very restricted set of allowed values (probably just "journal" and "proceedings"), like this:

```xml
<volume id="1" type="journal">
  ...
```

But of course we could also imagine this:

```xml
<volume id="1">
  <meta>
    <type>journal</type>
    ...
```

- - - 

So bottom line, this is a bit more complex than I thought :)

**Comment by mjpost** [comment]
1. `<journal-volume>` and `<journal-issue>` seem good to me
2. I think the book title is what's printed on the book and we should keep that, in part because it's too difficult to reconstruct it in a general way. I don't see the need to separate it out into two parts at the moment. I wonder if we need a separate `<journal-name>` tag or, maybe better, we retrieve that from `venues.yaml` using the Anthology ID? One complication is if a journal changes its name; so maybe we use `<journal-name>` if present but default to the venue name?
3. I like using the attribute to mark journals, and my inclination is that it would be good to be explicit and use an attribute for "proceedings" as you suggest here (so in the spec, the attribute is required)

**Comment by mbollmann** [comment]
In fact, we have both the journal name and the indication that it's a journal in the YAML files already, see e.g.:

https://github.com/acl-org/acl-anthology/blob/8b1510a0ce15194658651cbbd3b7d50330080ef1/data/yaml/venues/tacl.yaml#L1-L6

However, we've moved to represent venues explicitly in the XML using the `<venue>` tag, and there's nothing stopping a volume from being associated with both a journal and a non-journal venue. This is not currently happening in our data, but I wouldn't rule it out for the future (e.g., a workshop having a special issue in a journal).

Attaching at least the `type="journal"` to the `<volume>` directly seems like the cleanest solution to me; the journal title could still be retrieved from the YAML file by default if there's only one venue, or given explicitly in the XML otherwise, as @mjpost suggests. There's also one journal in our data already that changed names: CL, which was "American Journal of Computational Linguistics" until 1983.

**Comment by mbollmann** [comment]
Suggestion pushed to #2603, let's see if this builds and produces the correct result, particularly for the journal pages.

**Comment by mbollmann** [comment]
Going through the RelaxNG schema, I noticed some weird tags I didn't remember we had.

- - - 

_Papers_ (not volumes!) can currently have an `<issue>` tag. This is used mainly for LILT, although not consistently. Example:
https://github.com/acl-org/acl-anthology/blob/59a0926d74130da290abecf92e8ab863e40a3174/data/xml/2014.lilt.xml#L101-L130

I assume this was done to avoid creating many individual `<volume>`s containing only one paper each, although if that's how they should be cited (= as individual issues containing that one paper), that's probably how they should be represented?

By "not consistently" I mean that, for example, 2014.lilt-10 and 2014.lilt-11 have these tags, but 2014.lilt-9 does not.

They are [currently _not_ reflected in the BibTeX](https://aclanthology.org/2014.lilt-10.1) for these papers.

- - - 

_Papers_ can currently have a  `<journal>` tag. This is used in only one instance: EAMT 2016, where every single paper has:
https://github.com/acl-org/acl-anthology/blob/59a0926d74130da290abecf92e8ab863e40a3174/data/xml/W16.xml#L6770-L6771

Again, this is [not currently reflected in the BibTeX](https://aclanthology.org/W16-3401/).

We should get rid of this. Either the paper belongs to a proceedings volume, or a journal — the citation can't reflect both at the same time. We could add the "Baltic Journal of Modern Computing" as a venue and add the respective `<venue>` tag to this proceedings volume; or we could record the information that this was also published in this journal in the generic `<note>` field.
